lib/Core/Search/Common/EventSubscriber/ContentEventSubscriber.php

```diff
@@ -55,26 +59,24 @@ public function onCopyContent(CopyContentEvent $event): void
 
     public function onBeforeDeleteContent(BeforeDeleteContentEvent $event): void
     {
+        $contentLocations = $this->locationService->loadLocations($event->getContentInfo());
         try {
-            $event->getContentInfo()->getMainLocation()?->parentLocationId;
+            foreach ($contentLocations as $contentLocation){
+                $this->contentParentLocations[] = $contentLocation->parentLocationId;
+            }
         } catch (Throwable) {
             // does nothing
         }
     }
 
     public function onDeleteContent(DeleteContentEvent $event): void
     {
-        try {
-            $mainLocationParentLocationId = $event->getContentInfo()->getMainLocation()?->parentLocationId;
-        } catch (Throwable) {
-            $mainLocationParentLocationId = null;
-        }
-
+        $parentLocationIds = $this->contentParentLocations ?? [];
         $this->messageBus->dispatch(
             new DeleteContent(
                 $event->getContentInfo()->id,
                 $event->getLocations(),
-                $mainLocationParentLocationId,
+                $parentLocationIds,
```

Comment:
You can inline the value here.

Reply:
resolved in c7788685c4049ca5a31a9ae450be1ee56b04cef8